lib/fastlane/plugin/wpmreleasetoolkit/actions/common/check_for_toolkit_updates_action.rb

```diff
@@ -0,0 +1,77 @@
+require 'fastlane/action'
+require 'rubygems/command_manager'
+
+module Fastlane
+  module Actions
+    class CheckForToolkitUpdatesAction < Action
+      TOOLKIT_SPEC_NAME = 'fastlane-plugin-wpmreleasetoolkit'.freeze
+
+      def self.run(params)
+        updater = Gem::CommandManager.instance[:update]
+        installed_gems = updater.highest_installed_gems.select { |spec| spec == TOOLKIT_SPEC_NAME }
+        local_version = Gem::Version.new(installed_gems[TOOLKIT_SPEC_NAME].version)
+        UI.message("Currently using release toolkit version #{local_version.to_s.yellow}.")
+        UI.message('Checking for updates now...')
+        updates_needed = updater.which_to_update(installed_gems, [TOOLKIT_SPEC_NAME])
+
+        if updates_needed.empty?
+          UI.success('Your release toolkit is up-to-date! ✅')
+          return nil
+        end
+
+        # Return type of which_to_update differs before/after RubyGems 3.1.0, so normalize to always use Gem::NameTuple
+        updates_needed = Gem::NameTuple.from_list(updates_needed)
+        latest_version = updates_needed.find { |gem_info| gem_info.name == TOOLKIT_SPEC_NAME }.version
+
+        UI.message(['There is a new version '.yellow, latest_version.to_s.red, ' of the release toolkit!'.yellow].join)
+        non_breaking_update = Gem::Requirement.new(local_version.approximate_recommendation).satisfied_by?(latest_version)
+        UI.important('The latest version introduces breaking changes!') unless non_breaking_update
+
+        return latest_version if params[:skip_update_suggestion] || !UI.confirm('Do you want to run bundle update now?')
+
+        sh('bundle', 'update', TOOLKIT_SPEC_NAME)
+        UI.abort_with_message!("#{TOOLKIT_SPEC_NAME} have been updated. Please check and commit the changes in your Gemfile.lock file, then restart your previous invocation of fastlane to use the new toolkit.")
+      end
+
+      def self.description
+        'Check that we are on the latest version of the release toolkit, and propose to update if not'
+      end
+
+      def self.authors
+        ['Olivier Halligon']
+      end
+
+      def self.return_value
+        'Returns the latest version of the toolkit available if your toolkit is not up-to-date, or nil if you are already up-to-date.'
```

Comment:
Unless I am misunderstaning the [return statement](https://github.com/wordpress-mobile/release-toolkit/pull/269/files?file-filters%5B%5D=.md&file-filters%5B%5D=.rb#diff-3a08f262f19672eb0b3baa7c9336cb4ec676e8762a6f2905e6f9d8e831b9115aR30) it doesn't look like it's returning anything if `skip_update_suggestion` param is `false`. It also looks like it's returning the latest version value even if toolkit is up to date. Not sure it's worth it, but should we clarify those and maybe say:

```suggestion
        'Returns the latest version of the toolkit available if update is skipped, otherwise returns nil'
```

Or a better phrasing of this ^

Reply:
There's a `return nil` earlier in the code [on line 19](https://github.com/wordpress-mobile/release-toolkit/pull/269/files#diff-3a08f262f19672eb0b3baa7c9336cb4ec676e8762a6f2905e6f9d8e831b9115aR19) if there was no update found tho.

So:
 - If NO update is found (i.e. we are up to date), line 19 returns nil.
 - If an update is found but we don't update (either because we have skip_update_suggestion set, or because we were suggested but replied no to the `UI.confirm` prompt), line 30 returns the new version
 - If an update is found but we DO update, we end fastlane – with an `abort_with_message!` – after the `bundle update` is done and the abort message suggest the user to restart fastlane. Since we abort, there's no concept of returned value there.

Reply:
🤦